src/main/java/simplifier/services/LinkServiceImpl.java

```diff
@@ -95,6 +95,23 @@ public LinkGetterDto saveLink(LinkCreationDto linkCreationDto) {
                 .map(link -> linkMapper.linkToLinkDto(link));
     }
 
+    @Override
+    public LinkGetterDto updateLink(String shortened, LinkCreationDto link) {
+        Link updatedLink = linkMapper.linkDtoToLink(link);
+        Link savedLink = linkRepository.findByShortenedLink(shortened).get();
```

Comment:
In that case it is possible to pass non-existing shortened url and app will crash. It would be better to return 404 if link not found by shortened. Also add integration tests for that case